django/template/loader_tags.py

```diff
@@ -249,6 +250,39 @@ def do_block(parser, token):
     return BlockNode(block_name, nodelist)
 
 
+def construct_relative_path(name, relative_name):
+
+    if not any(map(lambda x: relative_name.startswith(x), [
+        "'./",
+        "'../",
+        '"./',
+        '"../',
+    ])):
+        # argument is variable or literal, that not contain relative path
+        return relative_name
+
+    new_name = posixpath.normpath(
+        posixpath.join(
+            posixpath.dirname(name.lstrip('/')),
+            relative_name.strip('\'"')
+        )
+    )
+
+    if new_name.startswith('../'):
+        raise TemplateSyntaxError(
+            "Relative name '%s' has more parent folders, then the template name %s in which the tag appears"
```

Comment:
English wash:

"Relative name '%s' points outside the file hierarchy that template %s is in"